src/openforms/formio/components/custom.py

```diff
@@ -639,6 +644,139 @@ def as_json_schema(component: AddressNLComponent) -> JSONObject:
         return base
 
 
+class PartnerListSerializer(serializers.ListSerializer):
+    def __init__(self, *args, **kwargs):
+        self.component = kwargs.pop("component", None)
+        super().__init__(*args, **kwargs)
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+
+        component_key = self.component["key"]
+        submission = self.context["submission"]
+        prefill_data = submission.get_prefilled_data()
+
+        fm_immutable_variables = FormVariable.objects.filter(
+            source=FormVariableSources.user_defined,
+            prefill_plugin=FM_PLUGIN_IDENTIFIER,
+        ).exclude(prefill_options={})
+
+        for immutable_var in fm_immutable_variables:
+            if (
+                immutable_var.prefill_options["mutable_data_form_variable"]
+                != component_key
+            ):
+                continue
+
+            initial_value = prefill_data[immutable_var.key]
+
+            # we do not send these information from the frontend (since they are not used
+            # until for now) so we have to exclude them from the data that needs validation
+            for partner in initial_value:
+                for key in ("dateOfBirthPrecision", "firstNames"):
+                    partner.pop(key, None)
```

Comment:
this looks very risky for accidental mutation, I think I'd feel safer with a coy of the data:

```py
initial_value = [
    {key: value for key, value in partner.items() if key not in ("dateOfBirthPrecision", "firstNames")}
    for partner in prefill_data[immutable_var.key]
]
```